app/views/directives/edit-environment-from.html

```diff
@@ -0,0 +1,94 @@
+      ng-repeat="entry in $ctrl.envFromEntries"
+      as-sortable-item>
+
+      <div
+        class="form-group key-value-editor-input"
+        ng-class="{ 'has-error': (forms.editEnvironmentFrom[uniqueForValue(unique, $index)].$invalid && forms.editEnvironmentFrom[uniqueForValue(unique, $index)].$touched) }">
+        <div ng-if="$ctrl.isEnvFromReadonly(entry)" class="faux-input-group">
+          <div class="faux-form-control readonly">
+            Set to values in {{entry.selectedEnvFrom.kind | humanizeKind : true | lowercase}}
+            <span
+              ng-if="!('configmaps' | canI : 'get') || !('secrets' | canI : 'get')">
+              {{entry.configMapRef.name || entry.secretRef.name}}
+            </span>
+            <a
+              ng-if="'configmaps' | canI : 'get'"
+              ng-href="{{entry.selectedEnvFrom | navigateResourceURL}}">
+              {{entry.configMapRef.name || entry.secretRef.name}}
+            </a>
+          </div>
+        </div>
+
+        <div ng-if="!$ctrl.isEnvFromReadonly(entry)">
+          <div class="ui-select">
+            <ui-select ng-model="entry.selectedEnvFrom"
+                       ng-required="entry.selectedEnvFrom"
+                       on-select="$ctrl.envFromObjectSelected($index, entry, $select.selected)"
+                       ng-class="{'{{$ctrl.setFocusClass}}' : $last}">
+              <ui-select-match placeholder="Select a resource">
+                <span>
+                  {{$select.selected.metadata.name}}
+                  <small class="text-muted">&ndash; {{$select.selected.kind | humanizeKind : true}}</small>
+                </span>
+              </ui-select-match>
+              <ui-select-choices
+                ui-disable-choice="$ctrl.checkEntries(source)"
+                repeat="source in $ctrl.envFromSelectorOptions | filter : { metadata: { name: $select.search } } track by (source | uid)"
+                group-by="$ctrl.groupByKind">
+                <span ng-bind-html="source.metadata.name | highlight : $select.search"></span>
+              </ui-select-choices>
+            </ui-select>
+          </div>
+        </div>
+
+        <div>
+          <span
+            ng-if="(!$ctrl.cannotSort) && ($ctrl.entries.length > 1)"
+            class="fa fa-bars sort-row"
+            role="button"
+            aria-label="Move row"
+            aria-grabbed="false"
+            as-sortable-item-handle></span>
+          <a
+            href=""
+            class="pficon pficon-close delete-row as-sortable-item-delete"
+            role="button"
+            aria-label="Delete row"
+            ng-hide="$ctrl.cannotDeleteAny"
+            ng-click="$ctrl.deleteEntry($index, 1)"></a>
+          <a
+            ng-href="{{entry.selectedEnvFrom | navigateResourceURL}}"
+            class="pficon"
+            ng-show="entry.selectedEnvFrom"
+            ng-click="$ctrl.viewDetail(entry)">View {{entry.selectedEnvFrom.kind | humanizeKind : true}}</a>
```

Comment:
I think `$ctrl.viewDetail` doesn't exist?  the href works, tho.

Reply:
Yeahhhhhhhh =), removing appears superfluous looks like it was from a previous iteration